daemon/algod/api/server/v2/utils.go

```diff
@@ -430,15 +431,26 @@ func convertProgramTrace(programTrace []simulation.OpcodeTraceUnit) *[]model.Sim
 	return &modelProgramTrace
 }
 
+func digestOrNil(digest crypto.Digest) *string {
+	if digest.IsZero() {
+		return nil
+	}
+	digestStringFromBytes := string(digest.ToSlice())
+	return &digestStringFromBytes
+}
+
 func convertTxnTrace(txnTrace *simulation.TransactionTrace) *model.SimulationTransactionExecTrace {
 	if txnTrace == nil {
 		return nil
 	}
 
 	execTraceModel := model.SimulationTransactionExecTrace{
 		ApprovalProgramTrace:   convertProgramTrace(txnTrace.ApprovalProgramTrace),
+		ApprovalProgramHash:    digestOrNil(txnTrace.ApprovalProgramHash),
```

Comment:
This should be []byte, not a string...

Reply:
BlockHash is a string

Reply:
I converted the digest to a base32 encoded string in rest response for uniformity tho.